generators/module/index.js

```diff
@@ -1,19 +1,23 @@
 const R = require('ramda');
 const Base = require('../../lib/Base');
 
-
 module.exports = class extends Base {
-
-  get prompts() {
-    return ['modulesDir', 'moduleName'];
+  get requiredProps() {
+    return ['moduleName', 'moduleNameUpperFirst', 'modulePath'];
   }
 
   prompting() {
     return super.prompting();
   }
 
   writing() {
-    R.forEach(this.copy)([
+    R.forEach((file) => {
+      this.fs.copyTpl(
+        this.templatePath(file),
+        this.destinationPath(`${this.props.modulePath}/${file}`),
+        R.pick(['moduleName', 'moduleNameUpperFirst'], this.props)
+      );
+    }, [
       'components/index.js',
```

Comment:
I don't disagree. In this particular case, there's nothing lost since the instance isn't calling `writing` ever more than once. I concede that to not be a great reason for doing anything, but is why I don't intend to make a change.